bitcoin/psbt.c

```diff
@@ -441,6 +442,47 @@ bool psbt_input_set_redeemscript(struct wally_psbt *psbt, size_t in,
 	return wally_err == WALLY_OK;
 }
 
+void psbt_input_hash_for_sig(struct wally_psbt *psbt, size_t in,
+			     struct sha256_double *dest)
+{
+       int ret;
+       u8 value[9];
+       u64 input_val_sats;
+       struct amount_sat input_amt;
+       int flags = WALLY_TX_FLAG_USE_WITNESS;
+
+       input_amt = psbt_input_get_amount(psbt, in);
+       input_val_sats = input_amt.satoshis; /* Raw: type conversion */
+
+       /* Wally can allocate here, iff tx doesn't fit on stack */
+       tal_wally_start();
+       if (is_elements(chainparams)) {
```

Comment:
This should use `wally_psbt_get_input_signature_hash` instead of calculating the hash itself from psbt internals. In particular, the wally call handles taproot for you, which this doesn't, and it will correctly match missing sighashes to `SIGHASH_DEFAULT`/`SIGHASH_ALL` for you (this code will just produce the wrong hash).